content/en/docs/reference/glossary/kubelet-eviction.md

```diff
@@ -0,0 +1,26 @@
+---
+title: Out-of-resource eviction
+id: kubelet-eviction
+date: 2021-04-27
+full_link: /docs/concepts/scheduling-eviction/pod-eviction/#kubelet-eviction
+short_description: >
+  Out-of-resource eviction is the process by the kubelet proactively fails pods
+  to reclaim resources.
+aka: kubelet eviction
+tags:
+- operation
+- fundamental
+---
+Out-of-resource eviction is the process by which pods are terminated to either
+reclaim resources for nodes, or to schedule higher-priority pods on a cluster.
+
+<!--more-->
+
+There are two types of out-of-resource eviction: 
+
+1. Cluster-level preemption: the scheduler evicts pods from nodes in a cluster
```

Comment:
Yes, have a separate entry for preemption.

Reply:
We already have a separate entry for preemption. I wonder if we should have link to that entry from the kubelet eviction entry, something like "To learn about cluster-level eviction, see preemption." Thoughts?

Reply:
Let's figure something out first:
Do we want to consider kubelet eviction and preemption both as types of "out-of-resource eviction"? If so, the title of this should have 3 entries: out-of-resource eviction, node eviction (or kubelet eviction) and cluster eviction (or preemption). The first references the other 2.

If not, we just have 2 entries: node eviction and cluster eviction.

@sftim what is your take?

Reply:
Are we married to calling it "out-of-resource eviction"? If this was called kubelet eviction, cluster-level called preemption, and the Eviction object one called graceful eviction, would that make it cleaner?

Reply:
> Let's figure something out first:
> Do we want to consider kubelet eviction and preemption both as types of "out-of-resource eviction"? If so, the glossary should have 3 entries: out-of-resource eviction, node eviction (or kubelet eviction) and cluster eviction (or preemption). The first references the other 2.
> 
> If not, we just have 2 entries: node eviction and cluster eviction.

I guess since preemption isn't really "out-of-resource" in the same way that node eviction is (at least the way I, a relative newbie, understand it), I'm in favour of just having them as node eviction and cluster eviction. I can do that in this PR and then if we decide we do want them grouped, we can just add that in a future PR.

Reply:
Also - node-level eviction (so that there's nothing to suggest that entire nodes are getting booped)?